ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemRootResourceDefinition.java

```diff
@@ -123,7 +123,6 @@
             new SimpleAttributeDefinitionBuilder(EJB3SubsystemModel.ENABLE_STATISTICS, ModelType.BOOLEAN, true)
                     .setAllowExpression(true)
                     .setDeprecated(ModelVersion.create(5))
-                    .setFlags(AttributeAccess.Flag.ALIAS)
```

Comment:
This attribute is an alias.

Reply:
This one is tricky. There is only one instance of ALIAS in the whole of Wildfly and its this one. I discussed with PF and RH and they said don't use it. If I do use it, the transformer tests fail. Is there some description somewhere of how to handle attributes with an ALIAS flag? The description above the definition says, "this attribute is an alias to something else". How are aliases handled in general?

Reply:
What is happening is the model comparison fails: in 6.0.0, statistics-enabled=expression and enable-statistics=expression; in 5.0.0, statistics-enabled=expression and enable-statistics=undefined. In 6.0.0, enable-statistics is flagged as ALIAS and there are custom read-attribute and write-attribute handlers for enable-statistics which instead read and write from statistics-enabled. This explains why the values are equal in 6.0.0. Exactly the same setup in 5.0.0 (EAP 7.2.0, 7.1.0). The operation transformers are creating the correct model, but the resource transformers are creating a model where enable-statistics is undefined. Is this a case where a value needs to be injected if it is undefined?

Reply:
I created an attribute converter for enable-statistics which will check if it is undefined and if it is, set the value to the value of statistics-enabled. I assumed that the undefined value was being created on the initial model transfer; this seems to fix it.

Reply:
You've revealed that a bug fix in WF 15 / EAP 7.3 (https://issues.redhat.com/browse/WFCORE-4183) fixes another problem besides the one it targeted. :-) When you invoke read-resource on an EAP 7.2 or earlier controller you get the value of aliases even though the op has the 'include-aliases="false" param set.

This is better to correct in DefaultModelFixer as opposed to changing the production code to send a value that the legacy host does not need. The result from the read-resource operation on the legacy controller is not correct due to that bug, so better to correct it in the test.

With https://github.com/bstansberry/wildfly/commit/fd3d27fe6cdf1f3fe453b72993026e35ac3c7c21 added to what you have here the EJB3 subsystem tests pass. Feel free to squash that in if you agree it is correct.

Reply:
Added in your code snippet to fix the issue in the ModelFixer. Thanks, Brian.